### PR TITLE
Couple of Zeek plugin updates.

### PR DESCRIPTION
**Pull request by rsmmr**
These go with https://github.com/zeek/spicy-analyzers

### CODE REVIEW

File: zeek/plugin/src/plugin.cc [hunk 1]
```diff
@@ -474,5 +481,36 @@ void plugin::Zeek_Spicy::Plugin::autoDiscoverModules() {
 
             globfree(&gl);
         }
+
+        // The following code is adapted from Zeek.
+        auto* d = opendir(dir.c_str());
```
Comment (bbannier) [inline zeek/plugin/src/plugin.cc:486]: `opendir` and `readdir` need `dirent.h`

File: zeek/plugin/src/plugin.cc [hunk 2]
```diff
@@ -474,5 +481,36 @@ void plugin::Zeek_Spicy::Plugin::autoDiscoverModules() {
...
+        if ( ! d )
+            return;
+
+        while ( auto dp = readdir(d) ) {
+            struct stat st;
+
+            if ( strcmp(dp->d_name, "..") == 0 || strcmp(dp->d_name, ".") == 0 )
```
Comment (bbannier) [inline zeek/plugin/src/plugin.cc:493]: Needs `cstring`.

File: tests/Scripts/have-zeek-plugin [hunk 3]
```diff
@@ -7,4 +7,4 @@ min_version=0
 test $# -ge 1 && min_version=$1
 
 spicy-config --zeek-plugin-path >/dev/null && \
-    test "$(spicy-config --zeek-version)" -ge "${min_version}"
+    test "$(spicy-config --zeek-version-number)" -ge "${min_version}"
```
Comment (bbannier) [inline tests/Scripts/have-zeek-plugin:10]: Yikes, the beauty of untyped programming in shell 😰

**Review by bbannier — CHANGES_REQUESTED** [review]

File: zeek/plugin/src/plugin.cc [hunk 4]
```diff
@@ -474,5 +481,36 @@ void plugin::Zeek_Spicy::Plugin::autoDiscoverModules() {
 
             globfree(&gl);
         }
+
+        // The following code is adapted from Zeek.
+        auto* d = opendir(dir.c_str());
+        if ( ! d )
+            return;
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply zeek/plugin/src/plugin.cc]: Yep, I caught this after the push and just fixed it already.

File: spicy/toolchain/bin/spicy-config.cc [hunk 5]
```diff
@@ -183,6 +191,19 @@ int main(int argc, char** argv) {
...
+        if ( opt == "--zeek-module-path" ) {
+#ifdef HAVE_ZEEK_PLUGIN
+            if ( hilti::configuration().uses_build_directory )
+                result.emplace_back(hilti::configuration().build_directory / "zeek/plugin/modules");
+            else
+                result.emplace_back(hilti::configuration().lib_directory / "spicy/Zeek_Spicy/modules");
+
+            continue;
+#else
+            exit(1);
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply spicy/toolchain/bin/spicy-config.cc:203]: Ok, I'm cleaning this up.

File: zeek/plugin/include/plugin.h [hunk 6]
```diff
@@ -222,6 +226,9 @@ class Plugin : public zeek::plugin::Plugin {
     };
 #endif
 
+    // History of modules requested to load so far.
+    std::set<hilti::rt::filesystem::path> _loaded_modules;
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply zeek/plugin/include/plugin.h:230]: I tend to use `set` if there's no reason to prefer the unordered version; it's shorter. :)

File: zeek/plugin/src/plugin.cc [hunk 7]
```diff
@@ -425,6 +425,17 @@ void plugin::Zeek_Spicy::Plugin::Done() {
...
+    // If our auto discovery ends up finding the same module multiple times,
+    // ignore subsequent requests.
+    auto canonical_path = hilti::rt::filesystem::canonical(path);
+
+    if ( _loaded_modules.find(canonical_path) != _loaded_modules.end() ) {
+        ZEEK_DEBUG(hilti::rt::fmt("Ignoring duplicate loading request for %s", path.native()));
+        return;
+    }
+
+    _loaded_modules.insert(canonical_path);
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply zeek/plugin/src/plugin.cc:437]: I'm realizing we don't even need to the new set if we just track the libraries by canonical path. I'll do that.

File: zeek/plugin/src/plugin.cc [hunk 8]
```diff
@@ -474,5 +481,36 @@ void plugin::Zeek_Spicy::Plugin::autoDiscoverModules() {
 
             globfree(&gl);
         }
+
+        // The following code is adapted from Zeek.
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply zeek/plugin/src/plugin.cc:485]: Oh, honestly, I didn't know about filesystem::directory_iterator. I'll use that.

**Review by bbannier — APPROVED** [review]

**Pull request merged**